database/legacy/v3/migrate.go

```diff
@@ -186,7 +193,43 @@ func (db *Migrator) insertTransactionMessages(tx types.TransactionRow, partition
 }
 
 func (db *Migrator) deleteOldMessagesByAddressFunction() error {
+	_, err := db.Sql.Exec("DROP FUNCTION IF EXISTS messages_by_address_old(text[],text[],bigint,bigint);")
+	return err
+}
+
+func (db *Migrator) modifyOldMessageByAddressFunction() error {
 	_, err := db.Sql.Exec("DROP FUNCTION IF EXISTS messages_by_address(text[],text[],bigint,bigint);")
+	if err != nil {
+		return err
+	}
+
+	_, err = db.Sql.Exec(`
+CREATE FUNCTION messages_by_address_old(
```

Comment:
Can I have some more context on why we need to have the `messages_by_addresses_old` and the `messages_by_addresses` existing at the same time?

Reply:
hi @RiccardoM because currently upon running `bdjuno migrate v3` the `transaction` & `message` will be renamed with `_old`, while the old `messages_by_addresses` function's body would not changed and it will use the newly created `transaction` & `message` tables which result in the func not working properly during the whole migration process.

So I thought we could either:
- drop the old func and create a new one right after altering old tables, and the new func will access the new tables(but limited data with only new coming blocks), or
- as this PR, make the old func still usable in parallel to the new func where users can access old and new data with both functions.

Reply:
I think the best thing to do  is to try achieving the first option: dropping the old `messages_by_address` function and re-creating it properly so that it can work with the new tables. That wouldn't require the UI to use a different query or risk them to not receive all the data

Reply:
With the previous logic, the `messages_by_address` function will no longer work after the table names are altered, because the old function will not use the `_old` tables but the new `transaction` & `message` tables. For example, I tried to call the psql function after the old tables renamed and new tables created, the query will fail: 
```sql
bdjuno=> SELECT messages_by_address_old('{sif1rt5llnthdkvshew5a7qu4dwq9r5kv8tur9g3kz}','{}',2,0);

ERROR:  column reference "height" is ambiguous
LINE 7: ORDER BY height DESC
                 ^
QUERY:  
SELECT message.transaction_hash, message.index, message.type, message.value, message.involved_accounts_addresses
FROM message
         JOIN transaction t on message.transaction_hash = t.hash
WHERE (cardinality(types) = 0 OR type = ANY (types))
  AND addresses && involved_accounts_addresses
ORDER BY height DESC
LIMIT "limit" OFFSET "offset"
```
the `message` and the `transaction` in the above query are new tables, which are used by old function and whose  schema does not comply.

Reply:
@huichiaotsou This was simply due to the fact that the old `messages_by_address` function used to join the `transaction` and `message` table to be able to sort the messages by height. Now that the `messages` table also has the `height` field, it was causing the ambiguity cause it wasn't able to determine which field (`transaction.height` or `message.height`) should be used during the sort. I've fixed this by migrating the `messages_by_address` function to not use any join and simply rely on the `message.height` field. I've just pushed this here. Let me know if this works (both migration and after if the values are returned properly).

Reply:
@RiccardoM tested and it worked properly 👍🏻